### PR TITLE
Populate SSH known_hosts file

### PR DESCRIPTION
**Pull request by rji**
This PR introduces the ability to add the public keys for known SSH hosts.
* On the Jenkins master, this is accomplished by passing a space-separated string of hosts using the environment variable `SSH_KNOWN_HOSTS`. `ssh-keyscan` will gather the public keys for each of these hosts and write them to `/etc/ssh/ssh_known_hosts`.
* On our default dind-agent Docker image for Jenkins build agents, I've added github.com specifically to the Dockerfile. Users not using GitHub should either add additional well-known hosts to this list, or adapt this Docker image to their organization's needs.

### CODE REVIEW

File: scripts/bootstrap.py [hunk 1]
```diff
@@ -69,6 +69,23 @@ def populate_jenkins_location_config(location_xml, host, port):
     tree.write(location_xml)
 
 
+def populate_known_hosts(hosts, dest_file):
+    """Gather SSH public key from one or more hosts and write out the
+    known_hosts file.
+
+    :param hosts: a string of hosts separated by whitespace
+    :param dest_file: the file to write the public keys to
```
Comment (colin-msphere) [inline scripts/bootstrap.py]: Please add whether `dest_file` should be an absolute or relative path.

File: scripts/bootstrap.py [hunk 2]
```diff
@@ -69,6 +69,23 @@ def populate_jenkins_location_config(location_xml, host, port):
...
+    """
+    dest_dir = '/'.join(dest_file.split('/')[0:-1])
```
Comment (colin-msphere) [inline scripts/bootstrap.py]: `os.path.join('/', dest_file.split('/')[0:-1])` ?

Comment (rji) [inline scripts/bootstrap.py]: I have no idea what I was thinking here; I could totally just use `os.path.dirname()`

File: scripts/bootstrap.py (hunk 1 above)
Comment (rji) [inline scripts/bootstrap.py]: Will do!

File: scripts/bootstrap.py (hunk 2 above)
Comment (jgehrcke) [inline scripts/bootstrap.py]: Love it!

**Comment by rji** [comment]
CI failed due to Docker Hub returning 500 during the image push:

```
Received unexpected HTTP status: 500 Internal Server Error
```

**Comment by sschneid** [comment]
:+1:

**Comment by ssk2** [comment]
I see this error when I attempt to run the new image:
```I0205 19:29:49.097610  7234 exec.cpp:134] Version: 0.27.0
I0205 19:29:49.101848  7247 exec.cpp:208] Executor registered on slave 2475592d-3f57-4306-9222-e2077e8b01ad-S0
Traceback (most recent call last):
  File "/usr/local/jenkins/bin/bootstrap.py", line 180, in <module>
    sys.exit(main())
  File "/usr/local/jenkins/bin/bootstrap.py", line 157, in main
    populate_known_hosts('/etc/ssh/ssh_known_hosts', ssh_known_hosts)
  File "/usr/local/jenkins/bin/bootstrap.py", line 108, in populate_known_hosts
    os.makedirs(dest_dir)
  File "/usr/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 2] No such file or directory: ''```

**Comment by rji** [comment]
Good catch, @ssk2. It appears I had the arguments to the function ordered incorrectly. Let me know if that looks better!

**Comment by ssk2** [comment]
Tested and working! Let's get some docs up here too :)

**Pull request merged**